src/client/views/pivot-application/collection-view-delegate/collection-view-delegate.tsx

```diff
@@ -154,10 +145,33 @@ export class CollectionViewDelegate {
     }, () => window.location.hash = '#' + dataCube.name);
   }
 
-  updateItem(collection: Collection, item: CollectionItem) {
+  updateCollection(collection: Collection): Q.Promise<any> {
+    const appSettings = this.getSettings();
+
+    return this.save(appSettings.addOrUpdateCollection(collection));
+  }
+
+  deleteCollection(collection: Collection): Q.Promise<any> {
+    const appSettings = this.getSettings();
+
+    const oldIndex = appSettings.collections.indexOf(collection);
+
+    const undo = () => {
+      this.setState({
+        appSettings: appSettings.addCollectionAt(collection, oldIndex)
```

Comment:
This does not undo it on the server. (The collection is still deleted if you refresh). This needs to call `this.save` also